pkg/cvo/update.go

```diff
@@ -139,3 +144,71 @@ func updateStatus(cvoClient versioned.Interface, updates []cincinnati.Update) er
 
 	return nil
 }
+
+func applyUpdate(cvoClient versioned.Interface, kubeClient typedv1.AppsV1Interface) {
+	config, err := getConfig(cvoClient)
+	if err != nil {
+		glog.Errorf("Failed to get CVO config: %v", err)
+		return
+	}
+	glog.V(4).Infof("Found CVO config: %s", config)
+
+	if config.DesiredUpdate.Payload == "" || config.DesiredUpdate.Version == "" || config.DesiredUpdate.Version == version.Version.String() {
+		return
+	}
+
+	cvo := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-version-operator",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"openshift-core": "cluster-version-operator",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32Ptr(1),
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxUnavailable: intOrStringPtr(intstr.FromInt(0)),
+					MaxSurge:       intOrStringPtr(intstr.FromInt(1)),
+				},
+			},
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"openshift-core": "cluster-version-operator",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"openshift-core": "cluster-version-operator",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:  "cluster-version-operator",
+						Image: config.DesiredUpdate.Payload,
+					}},
+					NodeSelector: map[string]string{
+						"node-role.kubernetes.io/master": "",
+					},
+					Tolerations: []corev1.Toleration{{
```

Comment:
```yaml
tolerations:
- effect: NoSchedule
   key: node-role.kubernetes.io/master
   operator: Exists
```

Reply:
I explicitly tolerate everything, just in case.